### DISCUSSION TITLE
Sonoff MiniR4 Switch

### DISCUSSION THREAD

**Discussion by digiservice2001** [RavenSystem/esp-homekit-devices#2069] [MEPLHAA Scripts]
Hi, 

Can you help me to convert this script from Shelly1L (ESP8266) to Sonoff MiniR4 (ESP32) please?

Shelly script: (works fine)
{"c":{"io":[[[0],3],[[13],6],[[4,14],6,0,1],[[5],2]],"l":0,"b":[{"g":13,"t":5}]},"a":[{"0":{"r":[{"g":5}]},"1":{"r":[{"g":5,"v":1}]},"b":[{"g":13},{"g":4},{"g":4,"t":0},{"g":14},{"g":14,"t":0}],"es":[{"t":22,"h":2,"n":5,"j":5,"y0":[{"v":75,"r":1,"0":{"m":[[-1]]}}]}]}]}

I tried to convert it to ESP32, but that's not ok, I can not add accessory to home app, I think the device doesn't connect to WiFi.
{"c":{"io":[[[19],3],[[0],6],[[27],6,0,1],[[26],2]],"l":19,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":26}]},"1":{"r":[{"g":26,"v":1}]},"b":[{"g":0},{"g":27},{"g":27,"t":0}],"es":[{"t":22,"h":2,"n":5,"j":5,"y0":[{"v":75,"r":1,"0":{"m":[[-1]]}}]}]}]}

GPIOs of the Sonoff MiniR4
LED: 19
Button: 0
Switch: 27
Relay: 26

I flashed Sonoff MiniR4 with fullhaaboot_esp32.bin. I don't know if it is the right version...
 
Can You help me to solve this script issue?

Thanx!

**Comment by leonrrafael** [comment]
Mini r4
{"c":{"io":[[[26,19],2],[[0],6]],"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":26}]},"1":{"r":[{"g":26,"v":1}]},"b":[{"g":0}]}]}

Need test

**Reply by philgood** [reply]
Mini R4 with manual switch:

{"c":{"io":[[[0,27],6,1],[[19,26],2]],"l":19,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":26}]},"1":{"r":[{"g":26,"v":1}]},"b":[{"g":0},{"g":27},{"g":27,"t":0}]}]}

Confirmed working.
